internal/util/typeutil/map.go

```diff
@@ -2,6 +2,16 @@ package typeutil
 
 import "sync"
 
+// MapEqual returns true if the two map contain the same keys and values
+func MapEqual[K comparable, V any](left, right map[K]V) bool {
+	for k, v := range left {
+		if v2, ok := right[k]; !ok || v != v2 {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
Only has `right contains left` logic.
Need unit tests for this important logic utility function.